src/PhpSpreadsheet/Writer/Xls/Worksheet.php

```diff
@@ -518,8 +518,10 @@ public function close()
                 foreach ($conditionalStyles as $conditional) {
                     if ($conditional->getConditionType() == \PhpOffice\PhpSpreadsheet\Style\Conditional::CONDITION_EXPRESSION
                         || $conditional->getConditionType() == \PhpOffice\PhpSpreadsheet\Style\Conditional::CONDITION_CELLIS) {
-                        if (!in_array($conditional->getHashCode(), $arrConditional)) {
-                            $arrConditional[] = $conditional->getHashCode();
+                        if (!isset($arrConditional[$conditional->getHashCode()])) {
+                            // This hash code has been handled
+                            $arrConditional[$conditional->getHashCode()] = true;
```

Comment:
This (old) code should make sure that $condinal is only written once, as I understand the code.